### PR TITLE
change input type=number in send and recieve amount

### PR DESCRIPTION
**Pull request by benalleng**
as referenced in #229 the inputs for where only numbers should be are actually text currently.  Hoping to prevent non-numbers and negative numbers  where  those would cause errors

### CODE REVIEW

File: frontend/src/routes/Receive.tsx [hunk 1]
```diff
@@ -16,11 +17,20 @@ function Receive() {
...
+            navigate('/')
+            toast("You can't receive nothing")
+            return
+        }
+        if (amount) {
         const params = objectToSearchParams<ReceiveParams>({ amount, description })
         // Important! Otherwise we might see a stale bip21 code
         queryClient.invalidateQueries({ queryKey: ['bip21'] })
         navigate(`/receive/qr?${params}`)
+          }
```
Comment (benthecarman) [inline frontend/src/routes/Receive.tsx:33]: this formatting looks weird, should run the linter

Comment (benalleng) [reply frontend/src/routes/Receive.tsx]: I just briefly went over the error handling again for the toast strings and I realized they refuse to display on the receive page. they work everywhere else but to see them on receive you must first navigate home

**Comment by benalleng** [comment]
I just briefly went over the error handling again for the toast strings and I realized they refuse to display on the receive page. they work everywhere else but to see them on receive you must first navigate home. I'll need to figure out why it is happening on this page before this pr is ready

**Comment by benalleng** [comment]
sorry for the wave of commits, I kept thinking of more and more edge cases...
It should now accept only numbers while allowing you to still create zero-amount invoices along with adding commas for when you need to keep track of typing larger numbers.
As well on phones it appears to be a number text field but still behaves like a text field so any non-numeric characters pasted in accidentally will still be treated with the same logic as a on desktop.
Shout out to chatGPT for coming in clutch.

**Review by futurepaul — APPROVED** [review]
nice, thank you! I have some followup changes so we can format this with underscores like in the ui (`100_000`) but this is a great start

**Pull request merged**